iroh-resolver/src/unixfs_builder.rs

```diff
@@ -617,20 +640,54 @@ pub enum AddEvent {
     },
 }
 
-pub async fn add_blocks_to_store<S: Store>(
+use async_stream::stream;
+
+fn add_blocks_to_store_chunked<S: Store>(
+    store: S,
+    mut blocks: Pin<Box<dyn Stream<Item = Result<Block>>>>,
+) -> impl Stream<Item = Result<AddEvent>> {
+    let mut chunk = Vec::new();
+    let mut chunk_size = 0u64;
+    const MAX_CHUNK_SIZE: u64 = 1024 * 1024 * 16;
+    let t = stream! {
+        while let Some(block) = blocks.next().await {
+            let block = block?;
+            let block_size = block.data().len() as u64;
+            let cid = *block.cid();
+            if chunk_size + block_size > MAX_CHUNK_SIZE {
+                tracing::info!("adding chunk of {} bytes", chunk_size);
+                store.put_many(chunk.clone()).await?;
+                chunk.clear();
+                chunk_size = 0;
+            } else {
+                chunk_size += block_size;
```

Comment:
aren’t you forgetting to store anything in the chunk?

Reply:
It would seem so... Maybe that is why it is that fast...